pkg/lib/serverTime.js

```diff
@@ -321,13 +295,49 @@
                     console.warn("failed to load time servers", error);
                     return result;
                 });
-    };
+    }
+
+    function set_custom_ntp_timesyncd(servers, enabled) {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
 
-    self.set_custom_ntp = function (servers, enabled) {
         const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${enabled ? "" : "#"}NTP=${servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    self.get_custom_ntp = function () {
+        /* We only support editing the configuration of
+         * systemd-timesyncd, by dropping a file into
+         * /etc/systemd/timesyncd.conf.d.  We assume that timesyncd is
+         * used when:
+         *
+         * - systemd-timedated is answering for
+         *   org.freedesktop.timedate1 as opposed to, say, timedatex.
+         *
+         * - systemd-timesyncd is enabled (false if chrony is being used)
+         *
+         * The better alternative would be to have an API in
+         * o.fd.timedate1 for managing the list of NTP server
+         * candidates.
+         */
+
+        if (!timedate1_service.exists || timedate1_service.unit.Id !== "systemd-timedated.service") {
+            console.log("systemd-timedated not in use, ntp server configuration not supported");
+            return Promise.resolve({ supported: false });
+        }
+
+        if (!timesyncd_service.enabled) {
+            console.log("systemd-timesyncd not enabled, ntp server configuration not supported");
+            return Promise.resolve({ supported: false });
+        }
+
+        return get_custom_ntp_timesyncd();
+    };
+
+    self.set_custom_ntp = function (servers, enabled) {
+        return set_custom_ntp_timesyncd(servers, enabled);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20221205-123635-07fec9b3-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)